lib/Rex/Commands/File.pm

```diff
@@ -543,6 +543,10 @@ sub file {
 
   }
   elsif ( exists $option->{"source"} && !$is_directory ) {
+    if ($need_md5) {
+      eval { $old_md5 = md5($file); };
+    }
```

Comment:
As you can see a few line below, this case is completely offloaded to the `upload` function. That does its own file identity checks via MD5 checksumming, and then includes it in the returned value whether it had to change the file or not.

I think checking the result of the upload function would be the correct way to handle this case, instead of having to calculate one more checksum.